primitives/src/parachain.rs

```diff
@@ -49,6 +49,25 @@ pub struct Candidate {
 	pub block: BlockData,
 }
 
+/// Candidate receipt type.
+#[derive(Debug, PartialEq, Eq, Clone, Serialize, Deserialize)]
+#[serde(rename_all = "camelCase")]
+#[serde(deny_unknown_fields)]
+pub struct CandidateReceipt {
+	/// The ID of the parachain this is a candidate for.
+	pub parachain_index: Id,
+	/// The collator's account ID
+	pub collator: ::Address,
```

Comment:
should be a signature

Reply:
the spec: https://github.com/w3f/polkadot-spec/blob/master/spec.md#candidate-parachain-block

Reply:
allows validators to ignore future candidates from a collator that has previously submitted invalid candidates (and, conversely, favour candidates from collators that have previously submitted valid blocks).

Reply:
is it a signature on the hash of the head-data? or of the digest of the candidate itself? in that case we'd have to include that digest in the receipt.